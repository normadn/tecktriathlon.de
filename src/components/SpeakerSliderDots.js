import React from "react";

const SpeakerSliderDots = (props) => (
  <div
    css={{
      marginLeft: "auto",
      marginRight: "auto",
      display: "grid",
      width: props.speakers.length * 20,
      gridTemplateColumns: props.speakers
        .map(() => "1fr ")
        .reduce((a, b) => a + b),
      paddingBottom: "16px"
    }}
  >
    {props.speakers.map((_, i) => (
      <button
        onClick={props.onClick(i)}
        key={i}
        css={{
          backgroundColor: i === props.currentIndex ? "#00205b" : "#aeb7c8",
          transition: "0.5s",
          borderRadius: "50%",
          marginLeft: "auto",
          marginRight: "auto",
          width: "6px",
          height: "6px",
          border: "none"
        }}
      />
    ))}
  </div>
);

export default SpeakerSliderDots;
