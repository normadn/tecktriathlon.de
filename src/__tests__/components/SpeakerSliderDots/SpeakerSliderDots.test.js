import React from "react";
import SpeakerSliderDots from "../../../components/SpeakerSliderDots";
import { shallow } from "enzyme";

const speakers = [
  {
    name: "Vinith Suriyakumar",
    title: "Founder and CEO of Clarifai",
    description:
      "Matthew is a machine learning Ph.D. and thought leader pioneering the field of applied artificial intelligence (AI). Matt's groundbreaking research in computer vision has propelled the image recognition industry from theory to real-world application. Matthew is a machine learning Ph.D. and thought leader pioneering the field of applied artificial intelligence (AI). Matt's groundbreaking research in computer vision has propelled the image recognition industry from theory to real-world application. Matthew is a machine learning Ph.D. and thought leader pioneering the field of applied artificial intelligence (AI). Matt's groundbreaking research in computer vision has propelled the image recognition industry from theory to real-world application. Matthew is a machine learning Ph.D. and ya.",
    image: "coolpic.png"
  },
  {
    name: "Noah Ifergan",
    title: "Founder and CEO of Clarifai",
    description:
      "Matthew is a machine learning Ph.D. and thought leader pioneering the field of applied artificial intelligence (AI). Matt's groundbreaking research in computer vision has propelled the image recognition industry from theory to real-world application. Matthew is a machine learning Ph.D. and thought leader pioneering the field of applied artificial intelligence (AI). Matt's groundbreaking research in computer vision has propelled the image recognition industry from theory to real-world application. Matthew is a machine learning Ph.D. and thought leader pioneering the field of applied artificial intelligence (AI). Matt's groundbreaking research in computer vision has propelled the image recognition industry from theory to real-world application. Matthew is a machine learning Ph.D. and ya.",
    image: "coolpic.png"
  }
];

describe("SpeakerSliderDots", () => {
  it("renders correctly", () => {
    expect(
      shallow(
        <SpeakerSliderDots speakers={speakers} onClick={(num) => () => {}} />
      )
    ).toMatchSnapshot();
  });
});
