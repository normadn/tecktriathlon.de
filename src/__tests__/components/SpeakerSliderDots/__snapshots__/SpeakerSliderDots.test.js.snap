// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SpeakerSliderDots renders correctly 1`] = `
<div
  css={
    Object {
      "display": "grid",
      "gridTemplateColumns": "1fr 1fr ",
      "marginLeft": "auto",
      "marginRight": "auto",
      "paddingBottom": "16px",
      "width": 40,
    }
  }
>
  <button
    css={
      Object {
        "backgroundColor": "#aeb7c8",
        "border": "none",
        "borderRadius": "50%",
        "height": "6px",
        "marginLeft": "auto",
        "marginRight": "auto",
        "transition": "0.5s",
        "width": "6px",
      }
    }
    key="0"
    onClick={[Function]}
  />
  <button
    css={
      Object {
        "backgroundColor": "#aeb7c8",
        "border": "none",
        "borderRadius": "50%",
        "height": "6px",
        "marginLeft": "auto",
        "marginRight": "auto",
        "transition": "0.5s",
        "width": "6px",
      }
    }
    key="1"
    onClick={[Function]}
  />
</div>
`;
